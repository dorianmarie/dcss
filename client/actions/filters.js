import { SET_FILTER_SCENARIOS_IN_USE } from './types';

export let setFilterScenariosInUse = scenariosInUse => dispatch => {
  dispatch({ type: SET_FILTER_SCENARIOS_IN_USE, scenariosInUse });
};
