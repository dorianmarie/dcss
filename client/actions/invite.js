import {
  GET_INVITES_ERROR,
  GET_INVITES_SUCCESS,
  SET_INVITE_ERROR,
  SET_INVITE_SUCCESS
} from './types';

export let getInvites = () => async dispatch => {
  try {
    const res = await (await fetch(`/api/invites`)).json();

    if (res.error) {
      throw res;
    }
    const { invites } = res;
    dispatch({ type: GET_INVITES_SUCCESS, invites });
    return invites;
  } catch (error) {
    dispatch({ type: GET_INVITES_ERROR, error });
    return null;
  }
};

export let setInvite = (id, params) => async dispatch => {
  try {
    const { status } = params;
    const updates = {};

    if (status) {
      updates.status = status;
    }

    const body = JSON.stringify(updates);
    const res = await (await fetch(`/api/invites/${id}`, {
      method: 'PUT',
      headers: {
        'Content-Type': 'application/json'
      },
      body
    })).json();

    if (res.error) {
      throw res;
    }

    const { invites } = res;
    const invite = invites.find(invite => invite.id === id);

    dispatch({ type: SET_INVITE_SUCCESS, invite });
    return invite;
  } catch (error) {
    dispatch({ type: SET_INVITE_ERROR, error });
    return null;
  }
};
