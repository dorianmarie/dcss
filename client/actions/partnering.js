import { GET_PARTNERING_ERROR, GET_PARTNERING_SUCCESS } from './types';

export let getPartnering = () => async (dispatch, getState) => {
  const state = getState();

  if (state.partnering.length) {
    return state.partnering;
  }

  try {
    const res = await (await fetch('/api/partnering')).json();

    if (res.error) {
      throw res;
    }

    const { partnering } = res;

    dispatch({ type: GET_PARTNERING_SUCCESS, partnering });
    return partnering;
  } catch (error) {
    dispatch({ type: GET_PARTNERING_ERROR, error });
    return null;
  }
};
