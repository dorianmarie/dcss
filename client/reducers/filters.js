import { combineReducers } from 'redux';
import { SET_FILTER_SCENARIOS_IN_USE } from '@actions/types';

export const scenariosInUse = (state = [], action) => {
  const { scenariosInUse, type } = action;

  if (!scenariosInUse) {
    return state;
  }

  switch (type) {
    case SET_FILTER_SCENARIOS_IN_USE: {
      return [...scenariosInUse];
    }
    default:
      return state;
  }
};

export default combineReducers({
  scenariosInUse
});
