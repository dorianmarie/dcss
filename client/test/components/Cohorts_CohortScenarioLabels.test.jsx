/** @TEMPLATE: BEGIN **/
import React from 'react';
jest.mock('react', () => ({
  ...jest.requireActual('react'),
  useLayoutEffect: jest.requireActual('react').useEffect
}));

import {
  fetchImplementation,
  mounter,
  reduxer,
  serialize,
  snapshotter,
  state
} from '../bootstrap';
import { unmountComponentAtNode } from 'react-dom';

import {
  act,
  fireEvent,
  prettyDOM,
  render,
  screen,
  waitFor
} from '@testing-library/react';
import userEvent from '@testing-library/user-event';

async function waitForPopper() {
  // Popper update() - https://github.com/popperjs/react-popper/issues/350
  await act(async () => await null);
}

/** @TEMPLATE: END **/

/** @GENERATED: BEGIN **/

import * as tlr from '@testing-library/react';

import { SET_FILTER_SCENARIOS_IN_USE } from '../../actions/types';
import * as filtersActions from '../../actions/filters';
jest.mock('../../actions/filters');

import * as QueryString from 'query-string';
jest.mock('query-string');

const s1 = {
  id: 1,
  title: 'Foo',
  description: 'A scenario about Foo!',
  status: 2,
  deleted_at: null,
  updated_at: '2020-12-10T17:50:19.074Z'
};
const s2 = {
  id: 2,
  title: 'Bar',
  description: 'A scenario about Bar!',
  status: 2,
  deleted_at: null,
  updated_at: '2020-12-10T17:50:19.074Z'
};
const s3 = {
  id: 3,
  title: 'Baz',
  description: 'A scenario about Baz!',
  status: 2,
  deleted_at: null,
  updated_at: '2020-12-10T17:50:19.074Z'
};

let cohort;
let scenarios;
let scenariosById;
let scenariosInUse;

import CohortScenarioLabels from '../../components/Cohorts/CohortScenarioLabels.jsx';
/** @GENERATED: END **/

/** @TEMPLATE: BEGIN **/
const original = JSON.parse(JSON.stringify(state));
let container = null;
let commonProps = null;
let commonState = null;
/** @TEMPLATE: END **/

beforeAll(() => {
  /** @TEMPLATE: BEGIN **/
  (window || global).fetch = jest.fn();
  /** @TEMPLATE: END **/
});

afterAll(() => {
  /** @TEMPLATE: BEGIN **/
  jest.restoreAllMocks();
  /** @TEMPLATE: END **/
});

beforeEach(() => {
  /** @TEMPLATE: BEGIN **/
  jest.useFakeTimers();
  container = document.createElement('div');
  container.setAttribute('id', 'root');
  document.body.appendChild(container);

  fetchImplementation(fetch);
  /** @TEMPLATE: END **/

  /** @GENERATED: BEGIN **/

  cohort = {
    id: 1,
    created_at: '2020-08-31T14:01:08.656Z',
    name: 'A New Cohort That Exists Within Inline Props',
    is_archived: false,
    runs: [
      {
        id: 11,
        user_id: 333,
        scenario_id: 99,
        created_at: '2020-03-28T19:44:03.069Z',
        updated_at: '2020-03-31T17:01:43.139Z',
        ended_at: '2020-03-31T17:01:43.128Z',
        consent_id: 8,
        consent_acknowledged_by_user: true,
        consent_granted_by_user: true,
        referrer_params: null,
        cohort_id: 1,
        run_id: 11
      }
    ],
    scenarios: [99],
    users: [
      {
        username: 'super',
        personalname: 'Super User',
        email: 'super@email.com',
        id: 999,
        roles: ['participant', 'super_admin'],
        is_anonymous: false,
        is_super: true,
        progress: {
          completed: [1],
          latestByScenarioId: {
            1: {
              description: '',
              is_run: true,
              is_complete: true,
              event_id: 1909,
              created_at: 1602454306144,
              generic: 'arrived at a slide.',
              name: 'slide-arrival',
              url: 'http://localhost:3000/cohort/1/run/99/slide/1'
            }
          }
        }
      },
      {
        username: 'facilitator',
        personalname: 'Facilitator User',
        email: 'facilitator@email.com',
        id: 555,
        roles: ['participant', 'facilitator', 'researcher', 'owner'],
        is_anonymous: false,
        is_super: false,
        is_owner: true,
        progress: {
          completed: [],
          latestByScenarioId: {
            1: {
              description: '',
              is_run: true,
              is_complete: false,
              scenario_id: 99,
              event_id: 1905,
              created_at: 1602454306144,
              generic: 'arrived at a slide.',
              name: 'slide-arrival',
              url: 'http://localhost:3000/cohort/1/run/99/slide/1'
            }
          }
        }
      },
      {
        username: 'researcher',
        personalname: 'Researcher User',
        email: 'researcher@email.com',
        id: 444,
        roles: ['participant', 'researcher'],
        is_anonymous: false,
        is_super: false,
        progress: {
          completed: [],
          latestByScenarioId: {
            1: {
              description: '',
              is_run: true,
              is_complete: false,
              scenario_id: 99,
              event_id: 1904,
              created_at: 1602454306144,
              generic: 'arrived at a slide.',
              name: 'slide-arrival',
              url: 'http://localhost:3000/cohort/1/run/99/slide/1'
            }
          }
        }
      },
      {
        username: 'participant',
        personalname: 'Participant User',
        email: 'participant@email.com',
        id: 333,
        roles: ['participant'],
        is_anonymous: false,
        is_super: false,
        progress: {
          completed: [],
          latestByScenarioId: {
            1: {
              description: '',
              is_run: true,
              is_complete: false,
              scenario_id: 99,
              event_id: 1903,
              created_at: 1602454306144,
              generic: 'arrived at a slide.',
              name: 'slide-arrival',
              url: 'http://localhost:3000/cohort/1/run/99/slide/1'
            }
          }
        }
      },
      {
        username: 'anonymous',
        personalname: '',
        email: '',
        id: 222,
        roles: ['participant'],
        is_anonymous: true,
        is_super: false,
        progress: {
          completed: [],
          latestByScenarioId: {
            1: {
              description: '',
              is_run: true,
              is_complete: false,
              scenario_id: 99,
              event_id: 1902,
              created_at: 1602454306144,
              generic: 'arrived at a slide.',
              name: 'slide-arrival',
              url: 'http://localhost:3000/cohort/1/run/99/slide/1'
            }
          }
        }
      }
    ],
    roles: ['super', 'facilitator'],
    usersById: {
      999: {
        username: 'super',
        personalname: 'Super User',
        email: 'super@email.com',
        id: 999,
        roles: ['participant', 'super_admin'],
        is_anonymous: false,
        is_super: true,
        progress: {
          completed: [1],
          latestByScenarioId: {
            1: {
              description: '',
              is_run: true,
              is_complete: true,
              event_id: 1909,
              created_at: 1602454306144,
              generic: 'arrived at a slide.',
              name: 'slide-arrival',
              url: 'http://localhost:3000/cohort/1/run/99/slide/1'
            }
          }
        }
      },
      555: {
        username: 'facilitator',
        personalname: 'Facilitator User',
        email: 'facilitator@email.com',
        id: 555,
        roles: ['participant', 'facilitator', 'researcher', 'owner'],
        is_anonymous: false,
        is_super: false,
        is_owner: true,
        progress: {
          completed: [],
          latestByScenarioId: {
            1: {
              description: '',
              is_run: true,
              is_complete: false,
              scenario_id: 99,
              event_id: 1905,
              created_at: 1602454306144,
              generic: 'arrived at a slide.',
              name: 'slide-arrival',
              url: 'http://localhost:3000/cohort/1/run/99/slide/1'
            }
          }
        }
      },
      444: {
        username: 'researcher',
        personalname: 'Researcher User',
        email: 'researcher@email.com',
        id: 444,
        roles: ['participant', 'researcher'],
        is_anonymous: false,
        is_super: false,
        progress: {
          completed: [],
          latestByScenarioId: {
            1: {
              description: '',
              is_run: true,
              is_complete: false,
              scenario_id: 99,
              event_id: 1904,
              created_at: 1602454306144,
              generic: 'arrived at a slide.',
              name: 'slide-arrival',
              url: 'http://localhost:3000/cohort/1/run/99/slide/1'
            }
          }
        }
      },
      333: {
        username: 'participant',
        personalname: 'Participant User',
        email: 'participant@email.com',
        id: 333,
        roles: ['participant'],
        is_anonymous: false,
        is_super: false,
        progress: {
          completed: [],
          latestByScenarioId: {
            1: {
              description: '',
              is_run: true,
              is_complete: false,
              scenario_id: 99,
              event_id: 1903,
              created_at: 1602454306144,
              generic: 'arrived at a slide.',
              name: 'slide-arrival',
              url: 'http://localhost:3000/cohort/1/run/99/slide/1'
            }
          }
        }
      },
      222: {
        username: 'anonymous',
        personalname: '',
        email: '',
        id: 222,
        roles: ['participant'],
        is_anonymous: true,
        is_super: false,
        progress: {
          completed: [],
          latestByScenarioId: {
            1: {
              description: '',
              is_run: true,
              is_complete: false,
              scenario_id: 99,
              event_id: 1902,
              created_at: 1602454306144,
              generic: 'arrived at a slide.',
              name: 'slide-arrival',
              url: 'http://localhost:3000/cohort/1/run/99/slide/1'
            }
          }
        }
      }
    },
    chat: {
      id: 1,
      scenario_id: null,
      cohort_id: 1,
      host_id: 999,
      created_at: '2020-12-08T21:51:33.659Z',
      updated_at: null,
      deleted_at: null,
      ended_at: null,
      users: [
        {
          username: 'super',
          personalname: 'Super User',
          email: 'super@email.com',
          id: 999,
          roles: ['participant', 'super_admin'],
          is_anonymous: false,
          is_super: true,
          progress: {
            completed: [1],
            latestByScenarioId: {
              1: {
                description: '',
                is_run: true,
                is_complete: true,
                event_id: 1909,
                created_at: 1602454306144,
                generic: 'arrived at a slide.',
                name: 'slide-arrival',
                url: 'http://localhost:3000/cohort/1/run/99/slide/1'
              }
            }
          }
        },
        {
          id: 4,
          username: 'credible-lyrebird',
          personalname: null,
          email: null,
          is_anonymous: true,
          single_use_password: false,
          roles: ['participant', 'facilitator'],
          is_super: false,
          updated_at: '2020-12-10T17:50:19.074Z',
          is_muted: false,
          is_present: true
        }
      ],
      usersById: {
        4: {
          id: 4,
          username: 'credible-lyrebird',
          personalname: null,
          email: null,
          is_anonymous: true,
          single_use_password: false,
          roles: ['participant', 'facilitator'],
          is_super: false,
          updated_at: '2020-12-10T17:50:19.074Z',
          is_muted: false,
          is_present: true
        },
        999: {
          username: 'super',
          personalname: 'Super User',
          email: 'super@email.com',
          id: 999,
          roles: ['participant', 'super_admin'],
          is_anonymous: false,
          is_super: true,
          progress: {
            completed: [1],
            latestByScenarioId: {
              1: {
                description: '',
                is_run: true,
                is_complete: true,
                event_id: 1909,
                created_at: 1602454306144,
                generic: 'arrived at a slide.',
                name: 'slide-arrival',
                url: 'http://localhost:3000/cohort/1/run/99/slide/1'
              }
            }
          }
        }
      }
    }
  };
  cohort.scenarios = [1, 2, 3];
  scenarios = [s1, s2, s3];
  scenariosById = scenarios.reduce((accum, record) => {
    accum[record.id] = record;
    return accum;
  }, {});

  scenariosInUse = [2];

  filtersActions.setFilterScenariosInUse = jest.fn();

  filtersActions.setFilterScenariosInUse.mockImplementation(
    scenariosInUse => async dispatch => {
      dispatch({ type: SET_FILTER_SCENARIOS_IN_USE, scenariosInUse });
      return scenariosInUse;
    }
  );

  delete window.location;
  // eslint-disable-next-line
  window.location = {
    search: ''
  };

  QueryString.parse.mockImplementation(() => {
    return {
      s: [1],
      search: ''
    };
  });

  /** @GENERATED: END **/

  /** @TEMPLATE: BEGIN **/
  commonProps = {};
  commonState = JSON.parse(JSON.stringify(original));
  /** @TEMPLATE: END **/
});

afterEach(() => {
  /** @TEMPLATE: BEGIN **/
  jest.runOnlyPendingTimers();
  jest.useRealTimers();
  jest.resetAllMocks();
  unmountComponentAtNode(container);
  container.remove();
  container = null;
  commonProps = null;
  commonState = null;
  /** @TEMPLATE: END **/
});

test('CohortScenarioLabels', () => {
  expect(CohortScenarioLabels).toBeDefined();
});

/** @GENERATED: BEGIN **/
test('Render 1 1', async done => {
  const Component = CohortScenarioLabels;
  const props = {
    ...commonProps,
    cohort
  };

  const state = {
    ...commonState
  };

  const ConnectedRoutedComponent = reduxer(Component, props, state);

  const { asFragment } = render(<ConnectedRoutedComponent {...props} />);
  expect(asFragment()).toMatchSnapshot();

  done();
});
/** @GENERATED: END **/

/* INJECTION STARTS HERE */

test('No scenarios in state', async done => {
  const Component = CohortScenarioLabels;

  const props = {
    ...commonProps,
    cohort
  };

  const filters = {
    scenariosInUse
  };

  const state = {
    ...commonState,
    filters,
    scenarios: [],
    scenariosById
  };

  const ConnectedRoutedComponent = reduxer(Component, props, state);

  const { asFragment } = render(<ConnectedRoutedComponent {...props} />);
  expect(asFragment()).toMatchSnapshot();

  done();
});

test('Selected scenariosInUse', async done => {
  const Component = CohortScenarioLabels;

  const props = {
    ...commonProps,
    cohort
  };

  const filters = {
    scenariosInUse
  };

  const state = {
    ...commonState,
    filters,
    scenarios,
    scenariosById
  };

  const ConnectedRoutedComponent = reduxer(Component, props, state);

  const { asFragment } = render(<ConnectedRoutedComponent {...props} />);
  expect(asFragment()).toMatchSnapshot();

  done();
});

test('No selected scenariosInUse, with unmatched search term', async done => {
  const Component = CohortScenarioLabels;

  QueryString.parse.mockImplementation(() => {
    return {
      s: [],
      search: 'z'
    };
  });

  const props = {
    ...commonProps,
    cohort
  };

  const filters = {
    scenariosInUse: []
  };

  const state = {
    ...commonState,
    filters,
    scenarios,
    scenariosById
  };

  const ConnectedRoutedComponent = reduxer(Component, props, state);

  const { asFragment } = render(<ConnectedRoutedComponent {...props} />);
  expect(asFragment()).toMatchSnapshot();

  done();
});

test('No selected scenariosInUse, no search term', async done => {
  const Component = CohortScenarioLabels;

  QueryString.parse.mockImplementation(() => {
    return {
      s: [],
      search: ''
    };
  });

  const props = {
    ...commonProps,
    cohort
  };

  const filters = {
    scenariosInUse: []
  };

  const state = {
    ...commonState,
    filters,
    scenarios,
    scenariosById
  };

  const ConnectedRoutedComponent = reduxer(Component, props, state);

  const { asFragment } = render(<ConnectedRoutedComponent {...props} />);
  expect(asFragment()).toMatchSnapshot();

  done();
});

test('Add and remove a filter', async done => {
  const Component = CohortScenarioLabels;

  const onChange = jest.fn();

  const props = {
    ...(commonProps || {}),
    cohort,
    onChange
  };

  const filters = {
    scenariosInUse
  };

  const state = {
    ...commonState,
    filters,
    scenarios,
    scenariosById
  };

  const ConnectedRoutedComponent = reduxer(Component, props, state);

  const { asFragment } = render(<ConnectedRoutedComponent {...props} />);
  expect(asFragment()).toMatchSnapshot();

  const buttons = await screen.getAllByRole('button');
  expect(asFragment()).toMatchSnapshot();

  userEvent.click(buttons[0]);
  expect(asFragment()).toMatchSnapshot();

  userEvent.click(buttons[1]);
  expect(asFragment()).toMatchSnapshot();

  userEvent.click(buttons[2]);
  expect(asFragment()).toMatchSnapshot();

  userEvent.click(buttons[2]);
  expect(asFragment()).toMatchSnapshot();

  userEvent.click(buttons[1]);
  expect(asFragment()).toMatchSnapshot();

  userEvent.click(buttons[0]);
  expect(asFragment()).toMatchSnapshot();

  done();
});

test('Add and remove a filter, at a page, with a search term', async done => {
  const Component = CohortScenarioLabels;

  QueryString.parse.mockImplementation(() => {
    return {
      page: 1,
      s: [],
      search: 'Foo'
    };
  });

  const onChange = jest.fn();

  const props = {
    ...(commonProps || {}),
    cohort,
    onChange
  };

  const filters = {
    scenariosInUse
  };

  const state = {
    ...commonState,
    filters,
    scenarios,
    scenariosById
  };

  const ConnectedRoutedComponent = reduxer(Component, props, state);

  const { asFragment } = render(<ConnectedRoutedComponent {...props} />);
  expect(asFragment()).toMatchSnapshot();

  const buttons = await screen.getAllByRole('button');
  expect(asFragment()).toMatchSnapshot();

  userEvent.click(buttons[0]);
  expect(asFragment()).toMatchSnapshot();

  userEvent.click(buttons[1]);
  expect(asFragment()).toMatchSnapshot();

  userEvent.click(buttons[2]);
  expect(asFragment()).toMatchSnapshot();

  userEvent.click(buttons[2]);
  expect(asFragment()).toMatchSnapshot();

  userEvent.click(buttons[1]);
  expect(asFragment()).toMatchSnapshot();

  userEvent.click(buttons[0]);
  expect(asFragment()).toMatchSnapshot();

  done();
});
