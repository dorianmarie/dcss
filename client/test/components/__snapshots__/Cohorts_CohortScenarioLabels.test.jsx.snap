// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Add and remove a filter 1`] = `
<DocumentFragment>
  <div
    class="ui labels"
  >
    <button
      aria-label="Click for cohorts that include Foo"
      class="ui small label"
      scenario="[object Object]"
      tabindex="0"
      value="1"
    >
      Foo
    </button>
    <button
      aria-label="Click for cohorts that include Bar"
      class="ui small label primary"
      scenario="[object Object]"
      tabindex="0"
      value="2"
    >
      Bar
    </button>
    <button
      aria-label="Click for cohorts that include Baz"
      class="ui small label"
      scenario="[object Object]"
      tabindex="0"
      value="3"
    >
      Baz
    </button>
  </div>
</DocumentFragment>
`;

exports[`Add and remove a filter 2`] = `
<DocumentFragment>
  <div
    class="ui labels"
  >
    <button
      aria-label="Click for cohorts that include Foo"
      class="ui small label"
      scenario="[object Object]"
      tabindex="0"
      value="1"
    >
      Foo
    </button>
    <button
      aria-label="Click for cohorts that include Bar"
      class="ui small label primary"
      scenario="[object Object]"
      tabindex="0"
      value="2"
    >
      Bar
    </button>
    <button
      aria-label="Click for cohorts that include Baz"
      class="ui small label"
      scenario="[object Object]"
      tabindex="0"
      value="3"
    >
      Baz
    </button>
  </div>
</DocumentFragment>
`;

exports[`Add and remove a filter 3`] = `
<DocumentFragment>
  <div
    class="ui labels"
  >
    <button
      aria-label="Click for cohorts that include Foo"
      class="ui small label primary"
      scenario="[object Object]"
      tabindex="0"
      value="1"
    >
      Foo
    </button>
    <button
      aria-label="Click for cohorts that include Bar"
      class="ui small label primary"
      scenario="[object Object]"
      tabindex="0"
      value="2"
    >
      Bar
    </button>
    <button
      aria-label="Click for cohorts that include Baz"
      class="ui small label"
      scenario="[object Object]"
      tabindex="0"
      value="3"
    >
      Baz
    </button>
  </div>
</DocumentFragment>
`;

exports[`Add and remove a filter 4`] = `
<DocumentFragment>
  <div
    class="ui labels"
  >
    <button
      aria-label="Click for cohorts that include Foo"
      class="ui small label primary"
      scenario="[object Object]"
      tabindex="0"
      value="1"
    >
      Foo
    </button>
    <button
      aria-label="Click for cohorts that include Bar"
      class="ui small label"
      scenario="[object Object]"
      tabindex="0"
      value="2"
    >
      Bar
    </button>
    <button
      aria-label="Click for cohorts that include Baz"
      class="ui small label"
      scenario="[object Object]"
      tabindex="0"
      value="3"
    >
      Baz
    </button>
  </div>
</DocumentFragment>
`;

exports[`Add and remove a filter 5`] = `
<DocumentFragment>
  <div
    class="ui labels"
  >
    <button
      aria-label="Click for cohorts that include Foo"
      class="ui small label primary"
      scenario="[object Object]"
      tabindex="0"
      value="1"
    >
      Foo
    </button>
    <button
      aria-label="Click for cohorts that include Bar"
      class="ui small label"
      scenario="[object Object]"
      tabindex="0"
      value="2"
    >
      Bar
    </button>
    <button
      aria-label="Click for cohorts that include Baz"
      class="ui small label primary"
      scenario="[object Object]"
      tabindex="0"
      value="3"
    >
      Baz
    </button>
  </div>
</DocumentFragment>
`;

exports[`Add and remove a filter 6`] = `
<DocumentFragment>
  <div
    class="ui labels"
  >
    <button
      aria-label="Click for cohorts that include Foo"
      class="ui small label primary"
      scenario="[object Object]"
      tabindex="0"
      value="1"
    >
      Foo
    </button>
    <button
      aria-label="Click for cohorts that include Bar"
      class="ui small label"
      scenario="[object Object]"
      tabindex="0"
      value="2"
    >
      Bar
    </button>
    <button
      aria-label="Click for cohorts that include Baz"
      class="ui small label"
      scenario="[object Object]"
      tabindex="0"
      value="3"
    >
      Baz
    </button>
  </div>
</DocumentFragment>
`;

exports[`Add and remove a filter 7`] = `
<DocumentFragment>
  <div
    class="ui labels"
  >
    <button
      aria-label="Click for cohorts that include Foo"
      class="ui small label primary"
      scenario="[object Object]"
      tabindex="0"
      value="1"
    >
      Foo
    </button>
    <button
      aria-label="Click for cohorts that include Bar"
      class="ui small label primary"
      scenario="[object Object]"
      tabindex="0"
      value="2"
    >
      Bar
    </button>
    <button
      aria-label="Click for cohorts that include Baz"
      class="ui small label"
      scenario="[object Object]"
      tabindex="0"
      value="3"
    >
      Baz
    </button>
  </div>
</DocumentFragment>
`;

exports[`Add and remove a filter 8`] = `
<DocumentFragment>
  <div
    class="ui labels"
  >
    <button
      aria-label="Click for cohorts that include Foo"
      class="ui small label"
      scenario="[object Object]"
      tabindex="0"
      value="1"
    >
      Foo
    </button>
    <button
      aria-label="Click for cohorts that include Bar"
      class="ui small label primary"
      scenario="[object Object]"
      tabindex="0"
      value="2"
    >
      Bar
    </button>
    <button
      aria-label="Click for cohorts that include Baz"
      class="ui small label"
      scenario="[object Object]"
      tabindex="0"
      value="3"
    >
      Baz
    </button>
  </div>
</DocumentFragment>
`;

exports[`Add and remove a filter, at a page, with a search term 1`] = `
<DocumentFragment>
  <div
    class="ui labels"
  >
    <button
      aria-label="Click for cohorts that include Foo"
      class="ui small label"
      scenario="[object Object]"
      tabindex="0"
      value="1"
    >
      Foo
    </button>
    <button
      aria-label="Click for cohorts that include Bar"
      class="ui small label primary"
      scenario="[object Object]"
      tabindex="0"
      value="2"
    >
      Bar
    </button>
    <button
      aria-label="Click for cohorts that include Baz"
      class="ui small label"
      scenario="[object Object]"
      tabindex="0"
      value="3"
    >
      Baz
    </button>
  </div>
</DocumentFragment>
`;

exports[`Add and remove a filter, at a page, with a search term 2`] = `
<DocumentFragment>
  <div
    class="ui labels"
  >
    <button
      aria-label="Click for cohorts that include Foo"
      class="ui small label"
      scenario="[object Object]"
      tabindex="0"
      value="1"
    >
      Foo
    </button>
    <button
      aria-label="Click for cohorts that include Bar"
      class="ui small label primary"
      scenario="[object Object]"
      tabindex="0"
      value="2"
    >
      Bar
    </button>
    <button
      aria-label="Click for cohorts that include Baz"
      class="ui small label"
      scenario="[object Object]"
      tabindex="0"
      value="3"
    >
      Baz
    </button>
  </div>
</DocumentFragment>
`;

exports[`Add and remove a filter, at a page, with a search term 3`] = `
<DocumentFragment>
  <div
    class="ui labels"
  >
    <button
      aria-label="Click for cohorts that include Foo"
      class="ui small label primary"
      scenario="[object Object]"
      tabindex="0"
      value="1"
    >
      Foo
    </button>
    <button
      aria-label="Click for cohorts that include Bar"
      class="ui small label primary"
      scenario="[object Object]"
      tabindex="0"
      value="2"
    >
      Bar
    </button>
    <button
      aria-label="Click for cohorts that include Baz"
      class="ui small label"
      scenario="[object Object]"
      tabindex="0"
      value="3"
    >
      Baz
    </button>
  </div>
</DocumentFragment>
`;

exports[`Add and remove a filter, at a page, with a search term 4`] = `
<DocumentFragment>
  <div
    class="ui labels"
  >
    <button
      aria-label="Click for cohorts that include Foo"
      class="ui small label primary"
      scenario="[object Object]"
      tabindex="0"
      value="1"
    >
      Foo
    </button>
    <button
      aria-label="Click for cohorts that include Bar"
      class="ui small label"
      scenario="[object Object]"
      tabindex="0"
      value="2"
    >
      Bar
    </button>
    <button
      aria-label="Click for cohorts that include Baz"
      class="ui small label"
      scenario="[object Object]"
      tabindex="0"
      value="3"
    >
      Baz
    </button>
  </div>
</DocumentFragment>
`;

exports[`Add and remove a filter, at a page, with a search term 5`] = `
<DocumentFragment>
  <div
    class="ui labels"
  >
    <button
      aria-label="Click for cohorts that include Foo"
      class="ui small label primary"
      scenario="[object Object]"
      tabindex="0"
      value="1"
    >
      Foo
    </button>
    <button
      aria-label="Click for cohorts that include Bar"
      class="ui small label"
      scenario="[object Object]"
      tabindex="0"
      value="2"
    >
      Bar
    </button>
    <button
      aria-label="Click for cohorts that include Baz"
      class="ui small label primary"
      scenario="[object Object]"
      tabindex="0"
      value="3"
    >
      Baz
    </button>
  </div>
</DocumentFragment>
`;

exports[`Add and remove a filter, at a page, with a search term 6`] = `
<DocumentFragment>
  <div
    class="ui labels"
  >
    <button
      aria-label="Click for cohorts that include Foo"
      class="ui small label primary"
      scenario="[object Object]"
      tabindex="0"
      value="1"
    >
      Foo
    </button>
    <button
      aria-label="Click for cohorts that include Bar"
      class="ui small label"
      scenario="[object Object]"
      tabindex="0"
      value="2"
    >
      Bar
    </button>
    <button
      aria-label="Click for cohorts that include Baz"
      class="ui small label"
      scenario="[object Object]"
      tabindex="0"
      value="3"
    >
      Baz
    </button>
  </div>
</DocumentFragment>
`;

exports[`Add and remove a filter, at a page, with a search term 7`] = `
<DocumentFragment>
  <div
    class="ui labels"
  >
    <button
      aria-label="Click for cohorts that include Foo"
      class="ui small label primary"
      scenario="[object Object]"
      tabindex="0"
      value="1"
    >
      Foo
    </button>
    <button
      aria-label="Click for cohorts that include Bar"
      class="ui small label primary"
      scenario="[object Object]"
      tabindex="0"
      value="2"
    >
      Bar
    </button>
    <button
      aria-label="Click for cohorts that include Baz"
      class="ui small label"
      scenario="[object Object]"
      tabindex="0"
      value="3"
    >
      Baz
    </button>
  </div>
</DocumentFragment>
`;

exports[`Add and remove a filter, at a page, with a search term 8`] = `
<DocumentFragment>
  <div
    class="ui labels"
  >
    <button
      aria-label="Click for cohorts that include Foo"
      class="ui small label"
      scenario="[object Object]"
      tabindex="0"
      value="1"
    >
      Foo
    </button>
    <button
      aria-label="Click for cohorts that include Bar"
      class="ui small label primary"
      scenario="[object Object]"
      tabindex="0"
      value="2"
    >
      Bar
    </button>
    <button
      aria-label="Click for cohorts that include Baz"
      class="ui small label"
      scenario="[object Object]"
      tabindex="0"
      value="3"
    >
      Baz
    </button>
  </div>
</DocumentFragment>
`;

exports[`No scenarios in state 1`] = `
<DocumentFragment>
  <div
    class="ui labels"
  >
    <button
      aria-label="Click for cohorts that include Foo"
      class="ui small label"
      scenario="[object Object]"
      tabindex="0"
      value="1"
    >
      Foo
    </button>
    <button
      aria-label="Click for cohorts that include Bar"
      class="ui small label primary"
      scenario="[object Object]"
      tabindex="0"
      value="2"
    >
      Bar
    </button>
    <button
      aria-label="Click for cohorts that include Baz"
      class="ui small label"
      scenario="[object Object]"
      tabindex="0"
      value="3"
    >
      Baz
    </button>
  </div>
</DocumentFragment>
`;

exports[`No selected scenariosInUse, no search term 1`] = `
<DocumentFragment>
  <div
    class="ui labels"
  >
    <button
      aria-label="Click for cohorts that include Foo"
      class="ui small label"
      scenario="[object Object]"
      tabindex="0"
      value="1"
    >
      Foo
    </button>
    <button
      aria-label="Click for cohorts that include Bar"
      class="ui small label"
      scenario="[object Object]"
      tabindex="0"
      value="2"
    >
      Bar
    </button>
    <button
      aria-label="Click for cohorts that include Baz"
      class="ui small label"
      scenario="[object Object]"
      tabindex="0"
      value="3"
    >
      Baz
    </button>
  </div>
</DocumentFragment>
`;

exports[`No selected scenariosInUse, with unmatched search term 1`] = `
<DocumentFragment>
  <div
    class="ui labels"
  >
    <button
      aria-label="Click for cohorts that include Foo"
      class="ui small label"
      scenario="[object Object]"
      tabindex="0"
      value="1"
    >
      Foo
    </button>
    <button
      aria-label="Click for cohorts that include Bar"
      class="ui small label"
      scenario="[object Object]"
      tabindex="0"
      value="2"
    >
      Bar
    </button>
    <button
      aria-label="Click for cohorts that include Baz"
      class="ui small label"
      scenario="[object Object]"
      tabindex="0"
      value="3"
    >
      Baz
    </button>
  </div>
</DocumentFragment>
`;

exports[`Render 1 1 1`] = `
<DocumentFragment>
  <div
    class="ui labels"
  >
    <div
      class="ui small label"
    >
      This cohort contains no scenarios.
    </div>
  </div>
</DocumentFragment>
`;

exports[`Selected scenariosInUse 1`] = `
<DocumentFragment>
  <div
    class="ui labels"
  >
    <button
      aria-label="Click for cohorts that include Foo"
      class="ui small label"
      scenario="[object Object]"
      tabindex="0"
      value="1"
    >
      Foo
    </button>
    <button
      aria-label="Click for cohorts that include Bar"
      class="ui small label primary"
      scenario="[object Object]"
      tabindex="0"
      value="2"
    >
      Bar
    </button>
    <button
      aria-label="Click for cohorts that include Baz"
      class="ui small label"
      scenario="[object Object]"
      tabindex="0"
      value="3"
    >
      Baz
    </button>
  </div>
</DocumentFragment>
`;
